### PR TITLE
Add support for the literal-Dict feature

### LINKED ISSUE #100: The literal-Dict feature isn't supported
Taking [the example from help](https://github.com/vim/vim/blob/f78b52ba24d8d294869cd013dcc4cd5d4a65cea7/runtime/doc/eval.txt#L540):

```vim
let mydict = #{zero: 0, one_key: 1, two-key: 2, 333: 3}
```

## Possible output syntax tree

```
(let_statement
  (identifier)
  (literal_dictionary
    (literal_dictionary_entry
      key: (literal_key)
      value: (integer_literal))
    (literal_dictionary_entry
      key: (literal_key)
      value: (integer_literal))
    (literal_dictionary_entry
      key: (literal_key)
      value: (integer_literal))
    (literal_dictionary_entry
      key: (literal_key)
      value: (integer_literal))))
```

Or as an output of tree-sitter's playground:

```
let_statement [0, 0] - [0, 55]
  identifier [0, 4] - [0, 10]
  literal_dictionary [0, 13] - [0, 55]
    literal_dictionary_entry [0, 15] - [0, 22]
      key: literal_key [0, 15] - [0, 19]
      value: integer_literal [0, 21] - [0, 22]
    literal_dictionary_entry [0, 24] - [0, 34]
      key: literal_key [0, 24] - [0, 31]
      value: integer_literal [0, 33] - [0, 34]
    literal_dictionary_entry [0, 36] - [0, 46]
      key: literal_key [0, 36] - [0, 43]
      value: integer_literal [0, 45] - [0, 46]
    literal_dictionary_entry [0, 48] - [0, 54]
      key: literal_key [0, 48] - [0, 51]
      value: integer_literal [0, 53] - [0, 54]
```

## Current output syntax tree

```
let_statement [0, 0] - [0, 3]
ERROR [0, 4] - [0, 55]
  identifier [0, 4] - [0, 10]
  keyword [0, 15] - [0, 19]
  keyword [0, 24] - [0, 31]
  keyword [0, 36] - [0, 39]
  keyword [0, 40] - [0, 43]
```

### PR DESCRIPTION
**Pull request by jchros**
Closes #100.

Notes:

- I didn't add tests for this because I didn't know where I was supposed to; I originally wanted to add tests on the file where the syntax for dictionaries are tested but there are no such tests.

- As the second commit states:
  When I ran `tree-sitter generate`, the generated code has been modified in surprising ways (e.g. some regexes have been modified).  I don't know if that's normal, so in doubt I put these changes in a separate commit.

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1482,6 +1483,15 @@ module.exports = grammar({
     dictionnary: ($) =>
       seq('{', commaSep($.dictionnary_entry), optional(','), '}'),
 
+    // :h literal-Dict
+    literal_dictionary_entry: ($) =>
+      seq(field('key', $.literal_key), ':', field('value', $._expression)),
```
Comment (vigoux) [inline grammar.js:1488]: Can you make that an unnamed node ?
```suggestion
    _literal_dictionary_entry: ($) =>
      seq(field('key', $.literal_key), ':', field('value', $._expression)),
```

**Review by vigoux — CHANGES_REQUESTED** [review]
Don't worry about the parser.c and parser.h, I will handle regenerating the parser to the correct ABI version in order to keep that stable.

If you want to bother with this though, please install the latest version of `tree-sitter`.

File: grammar.js (hunk 1 above)
Comment (jchros) [reply grammar.js:1488]: I think I can, but wouldn't that be inconsistent with the fact that `dictionnary_entry` nodes are named?

Comment (jchros) [reply grammar.js]: Done.

**Review by vigoux — CHANGES_REQUESTED** [review]
Now regarding tests, please add some test cases in `corpus/expressions.txt` please, and we will be definitely good to go !

**Review by vigoux — APPROVED** [review]

**Pull request merged**